Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -240,13 +243,67 @@ public NameValueCollection GetNavigationUrlParameters ()
   private bool? _enableSmartScrolling;
   private bool? _enableSmartFocusing;
   private readonly SmartPageClientScriptManager _clientScriptManager;
+  private bool? _isCspEnabled;
+  private bool? _isCspReportOnlyEnabled;
+  private readonly INonceGenerator _nonceGenerator;
+  private string? _cspNonceValue;
 
   public SmartPage ()
   {
     _smartPageInfo = new SmartPageInfo(this);
     _validatableControlInitializer = new ValidatableControlInitializer(this);
     _postLoadInvoker = new PostLoadInvoker(this);
     _clientScriptManager = new SmartPageClientScriptManager(base.ClientScript);
+    _nonceGenerator = new NonceGenerator();
+  }
+
+  protected override void OnPreRenderComplete (EventArgs e)
```

Comment:
please move down to after OnInit, possibly all the way down to before ProcessRequest